### DISCUSSION TITLE
Resetting Controller component not working

### DISCUSSION THREAD

**Discussion by javidjamae** [react-hook-form/react-hook-form#2941] [Q&A] [answered]
I took [the following example from the docs](https://codesandbox.io/s/react-hook-form-normalize-field-1jumn) which has a Controller wrapped currency component and I tried to modify it to reset the currency field, but the reset doesn't work.

[Here is my attempt](https://codesandbox.io/s/react-hook-form-normalize-field-forked-p5o2v?file=/src/index.js). 

Looking at [this other example](https://codesandbox.io/s/react-hook-form-controller-079xx), it seems like what I'm doing should work. I am using reset and I am passing in the default values into the function as recommended in the docs and in several posts, but I can't seem to get it to work. 

Any help on how to reset a component controlled with `<Controller>` would be greatly appreciated. 

The submit / reset buttons look like this:
```
<input type="submit" />
<input 
  style={{ display: "block", marginTop: 20 }} 
  type="reset" 
  value="Standard Reset" />
<input
  style={{ display: "block", marginTop: 20 }}
  type="button"
  onClick={() => form.reset(defaultValues)}
  value="Custom Reset"
/>
```

And the onSubmit looks like this:
```
const form = useForm({ defaultValues });

const onSubmit = (data) => {
  form.reset(defaultValues);
};
```

**Comment by bluebill1049** [comment, marked as answer]
look at this example: https://codesandbox.io/s/react-hook-form-v6-controller-qsd8r
and examples folder search for `parse` keyword

**Reply by javidjamae** [reply]
Thanks, but that's the example I mentioned that I already looked at. That example is a bit different than the one I'm referencing because it doesn't define an `onValueChange` property. 

If I remove that prop or comment out the code in the function, the reset seems to work.

This resets:
```
    <NumberFormat
      customInput={TextField}
      {...rest}
      value={currency}
      fullWidth
      thousandSeparator={true}
      decimalScale={2}
      isNumericString
      prefix="$ "
    />
```

This doesn't:
```
    <NumberFormat
      customInput={TextField}
      {...rest}
      value={currency}
      fullWidth
      thousandSeparator={true}
      decimalScale={2}
      onValueChange={(target) => {
        setCurrency(target.floatValue);
        onChange(target.floatValue * 100);
      }}
      isNumericString
      prefix="$ "
    />
```

The odd thing is that if I put a console.log in the `onValueChange` it doesn't seem to be called on the reset, so I'm not quite sure when / how it is called or why it is preventing the reset from working.

**Reply by bluebill1049** [reply]
where is the usage of `Controller`?

**Reply by javidjamae** [reply]
Sorry, I'm not understanding your question. Are you asking what line of my [codesandbox](https://codesandbox.io/s/react-hook-form-normalize-field-forked-p5o2v?file=/src/index.js) references the Controller component? 

There are two, starting on line 34:
```
        <Controller
          name="priceInCents"
          control={form.control}
          as={CurrencyFormat}
        />
        <label htmlFor="muiPriceInCents">Material UI Price</label>
        <Controller
          name="muiPriceInCents"
          control={form.control}
          as={MuiCurrencyFormat}
        />
```

**Reply by bluebill1049** [reply]
the CSB component is not working: https://codesandbox.io/s/react-hook-form-normalize-field-forked-p5o2v?file=/src/index.js:1997-2025

**Reply by bluebill1049** [reply]
can you take a look at this example: https://codesandbox.io/s/react-hook-form-parse-and-format-textarea-furtc?

**Reply by javidjamae** [reply]
I know it's not working, hence my question. :-) 

Like I mentioned in my original post, I merely took the [Format/Mask Field (Third Party Library)](https://codesandbox.io/s/react-hook-form-normalize-field-1jumn) example and added a reset as described in the docs. I didn't change the code in any other way. 

I'm basically just trying to figure out how to do a reset on the Format/Mask Field (Third Party Library) example. 

The reset doesn't work because the original example uses `onValueChange` to update the values dynamically every time the user types. Removing `onValueChange` allows the reset, but then the value no longer get updated in the state as the user types. 

The example you provided does not do a reset at all, so I'm not sure how it is related. 

Either way, thanks so much for your attention to my question!

**Reply by bluebill1049** [reply]
> The example you provided does not do a reset at all, so I'm not sure how it is related.

It's related to your question because the root causes of its Controller is not wired correctly with its props.

https://codesandbox.io/s/react-hook-form-normalize-field-forked-01lgs?file=/src/index.js

**Reply by javidjamae** [reply]
Got it, thanks so much!

It may be worth updating the original example then, if that is the right way to wire in the props.

**Reply by bluebill1049** [reply]
more than happy to get it updated, which csb?

**Reply by javidjamae** [reply]
[ Format/Mask Field (Third Party Library)](https://codesandbox.io/s/react-hook-form-normalize-field-1jumn)

![Screen Shot 2020-09-17 at 5 44 27 PM](https://user-images.githubusercontent.com/355931/93542327-8c2ef480-f90d-11ea-890c-37e896fcca2e.png)

**Reply by bluebill1049** [reply]
thanks will do.
